pdns/lua-record.cc

```diff
@@ -168,6 +171,7 @@ class IsUpOracle
   }
   void checkThread()
   {
+    setThreadName("pdns/luaupcheck");
```

Comment:
If I'm not mistaken, `std::async` doesn't guarantee that the task will be executed in a different thread (although it's very likely), and if it is the thread might be long-lived and reused for a different task (pool of threads). 
So perhaps we might want to execute `setThreadName("pdns/luaupcheck");` again after"joining" the futures, to be sure that the name is not misleading while we are sleeping?

Reply:
it's my understanding that the `::async` launch policy guarantees a thread (but that's more based on current implementations than specs), but your idea is very good, I'll do it.

Reply:
> it's my understanding that the `::async` launch policy guarantees a thread 

`as if in a new thread of execution represented by a std::thread object. ` Yup, you are right!

Reply:
did it anyway - what if we change the policy later, etc.